### ISSUE TITLE
MNN使用Core ML后端精度很低

### ISSUE THREAD

**Issue by emojilearning** [alibaba/MNN#2255]
MNN使用Core ML后端精度很低。

res-Unet结构
使用Metal后端结果正常，~100ms
使用CoreML结果数值上相差很大，~20ms

# 平台(如果交叉编译请再附上交叉编译目标平台):  
# Platform(Include target platform as well if cross-compiling):  
iOS

# Github版本:
# Github Version:
2.3.0

# 编译方式:
# Compiling Method
开启了相关选项

# 编译日志:
# Build Log:
无误
```
粘贴在这里
Paste log here or pastebin
```

**Comment by wangzhaode** [comment]
> 结果数值上相差很大

请问具体误差有多少？正常来说CoreML的精度与fp16计算精度相当

**Comment by emojilearning** [comment]
差了好几倍。大概某一个输出Metal后端是0.5，CoreML是1.2这样。

**Comment by wangzhaode** [comment]
模型能提供吗

**Comment by emojilearning** [comment]
没有模型，但是可以给两个复现的代码片
`

def positionalencodingxy(input):
    """
    :param d_model: dimension of the model
    :param height: height of the positions
    :param width: width of the positions
    :return: d_model*height*width position matrix
    """
    d_model, height, width = input.shape[1::]
    pe = torch.zeros(d_model, height, width,device = input.device)
    # Each dimension use half of d_model
    d_model = int(d_model / 2)
    pos_w = torch.arange(0., width).unsqueeze(1)
    pos_h = torch.arange(0., height).unsqueeze(1)
    pe[0:1, :, :] = (pos_w).transpose(0, 1).unsqueeze(1).repeat(1, height, 1)
    pe[1:2, :, :] = (pos_h).transpose(0, 1).unsqueeze(2).repeat(1, 1, width)

    return pe

`

·
def forward(x)
    return torch.cat([x[:,1:,:,:],x[:,0:1,:,:],dim=1)
·
